extensions/amp-access/0.1/amp-access.js

```diff
@@ -118,6 +118,12 @@ export class AccessService {
     /** @private {?Promise<string>} */
     this.readerIdPromise_ = null;
 
+    /** @private {!Promise} */
+    this.firstAuthorizationPromise_ = new Promise(resolve => {
```

Comment:
Instead of creating a another promise, why not just store the return of `#runAuthorization_`? That'll be a promise that either resolves on response, or rejects (and any onFulfilleds won't be called) on error.